src/docs/development/add-to-app/android/add-flutter-screen.md

```diff
@@ -205,30 +132,35 @@ public class MyApplication extends Application {
   Flutter content.
 {{site.alert.end}}
 
-With a pre-warmed `FlutterEngine`, you now need to instruct your
-`FlutterActivity` to use the pre-warmed `FlutterEngine` instead of creating a
-new one. To accomplish this, subclass `FlutterActivity` and override the
-`provideFlutterEngine()` method:
+With a pre-warmed, cached `FlutterEngine`, you now need to instruct your
+`FlutterActivity` to use the cached `FlutterEngine` instead of creating a
+new one. To accomplish this, use `FlutterActivity`'s `withCachedEngine()`
+builder:
 
 ```java
-public class MyFlutterActivity extends FlutterActivity {
+myButton.addOnClickListener(new OnClickListener() {
   @Override
-  @NonNull
-  protected FlutterEngine provideFlutterEngine(@NonNull Context context) {
-    return ((MyApplication) context.getApplicationContext()).getFlutterEngine();
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .withCachedEngine("my_engine_id")
+        .build(currentActivity)
+      );
   }
-}
+});
 ```
 
-Now, when you launch `MyFlutterActivity`, there is significantly less delay in
+When using the `withCachedEngine()` factory method, pass the same ID that you
+used when caching the desired `FlutterEngine`.
+
+Now, when you launch `FlutterActivity`, there is significantly less delay in
```

Comment:
Not specifically for this line but we should also make sure to not make it sound like performance is the only reason. e.g. other reasons could be to run dart code or connect plugins early, or to keep the engine alive after the activity etc